File: nagasenacs/src/Nagasena/Proc/IO/Compression/ScannerChannelFactory.cs
using Channel = Nagasena.Proc.Common.Channel;

namespace Nagasena.Proc.IO.Compression {

  internal sealed class ScannerChannelFactory : ChannelFactory {

    internal override Channel createChannel(int firstPos, int blockNum) {
      return new ScannerChannel(firstPos, blockNum);
    }

  }

}

File: sumerogi/src/com/sumerogi/proc/io/compression/ScannerChannelFactory.java
package com.sumerogi.proc.io.compression;

import com.sumerogi.proc.common.Channel;

final class ScannerChannelFactory extends ChannelFactory {

  @Override
  public Channel createChannel(int firstPos, int blockNum) {
    return new ScannerChannel(firstPos, blockNum);
  }

}
